examples_tests/41.VisibilityBuffer/shader.frag

```diff
@@ -0,0 +1,16 @@
+#version 460 core
+
+layout (location = 0) in vec3 LocalPos;
+layout (location = 1) in vec3 ViewPos;
+layout (location = 2) in vec3 Normal;
+layout (location = 3) in vec2 UV;
+
+layout(location = 0) out vec4 color;
+
+void main()
+{
+    vec3 colorTmp = clamp(dot(vec3(0.0, 1.0, 0.0), Normal), 0, 1) * vec3(1.0) + vec3(0.2);
+    color = vec4(colorTmp, 1.0);
+
+    //color = vec4(1.0);

```

Comment:
for visibility buffer, you want to just create an FBO with a depth attachment and a RG32_UINT attachment.

then using  a `flat uvec2` you want to pump `uvec2(gl_DrawID,gl_VertexIndex)` from the vertex stage to `layout(location = 0) out uvec2 visBuffer;`

Then you want to run a compute shader with the "stolen" original vertex and fragment shader logic (and the descriptor with virtual texture and packed geometry).

See example 22 for how a visibility buffer works.

You can ask @Crisspl and me for help